lib/pymedphys/_experimental/chartchecks/dvh_helpers.py

```diff
@@ -0,0 +1,62 @@
+# Copyright (C) 2020 Jacob Rembish
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from pymedphys._imports import dicompylercore
+from pymedphys._imports import pandas as pd
+from pymedphys._imports import plotly, plt
+from pymedphys._imports import streamlit as st
+
+
+def plot_dvh(rs_file, rd_file):
+    rs_file = (
+        "P:/Share/Chris/Test_Patient/RS.1.3.46.670589.13.15476.20191204085114.491830"
+    )
+    rd_file = (
+        "P:/Share/Chris/Test_Patient/RD.1.3.46.670589.13.15476.20191204085119.163981"
+    )
+    # rp_file = "P:/Share/Chris/Test_Patient/RP.1.3.46.670589.13.15476.20191204085117.752796"
+
+    rs = dicompylercore.dicomparser.DicomParser(rs_file)
+    # rd = dicomparser.DicomParser(rd_file)
+    # rp = dicomparser.DicomParser(rp_file)
+
+    structures = rs.GetStructures()
+    dvh_structures = {"name": [], "bincenters": [], "counts": [], "volume": []}
+    # traces = []
+    fig = plt.subplots()[0]
+    for i in range(1, len(structures) + 1):
+        calcdvh = dicompylercore.dvhcalc.get_dvh(rs_file, rd_file, i)
+        dvh_structures["name"].append(calcdvh.name)
+        dvh_structures["bincenters"].append(calcdvh.bincenters)
+        dvh_structures["counts"].append(calcdvh.counts)
+        dvh_structures["volume"].append(calcdvh.volume)
+    #     ax.plot(structures[i]['bincenters'], structures[i]['counts']/structures[i]['volume'], label = calcdvh.name)
+    #     ax.legend(bbox_to_anchor=(1.05, 1), loc='upper left', borderaxespad=0.)
+    # ax.set(ylim=(0,1))
+    # st.pyplot(fig)
+    data = pd.DataFrame.from_dict(dvh_structures)
+    data["rel_volume"] = data["counts"] / data["volume"]
+    d1 = data.explode("bincenters").drop(columns=["counts", "rel_volume"])
```

Comment:
Any ideas how to make my DVH plotting quicker? It was much faster when I used matplotlib, however I like using plotly as it gives me an interactive DVH (I can select which structures I want displayed, hover over to get values, etc). The main problem with plotly is (to my knowledge) I can't use an array within a dataframe cell. I have to use the explode function to give every bin it's own row, and I believe this is the main part slowing it down.

Reply:
I'd need to look into this quote a bit deeper to answer this one. If you could produce a mini Streamlit app, one with matplotlib and one with plotly and then post the code on https://discuss.streamlit.io they should be able to give you a nice answer quicker than I can :)